cpu/esp32/esp-ieee802154/esp_ieee802154_hal.c

```diff
@@ -0,0 +1,461 @@
+        return esp_ieee802154_add_pending_addr(value, false) ? -ENOMEM : 0;
+    case IEEE802154_SRC_MATCH_EXT_CLEAR:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_EXT_CLEAR\n", __func__);
+        return esp_ieee802154_clear_pending_addr(value, false) ? -ENOENT : 0;
+    }
+
+    return -EINVAL;
+}
+
+void esp_ieee802154_cca_done(bool channel_busy)
+{
+    DEBUG("[esp_ieee802154] %s %d\n", __func__, channel_busy);
+    _channel_clear = !channel_busy;
+
+    if (IS_ACTIVE(_USE_CCA_DONE)) {
+        esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_CONFIRM_CCA);
+    }
+}
+
+void esp_ieee802154_receive_sfd_done(void)
+{
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+    if (IS_ACTIVE(_USE_RX_START)) {
+        esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_INDICATION_RX_START);
+    }
+}
+
+void esp_ieee802154_receive_done(uint8_t *frame, esp_ieee802154_frame_info_t *frame_info)
+{
+    assert(frame);
+    assert(frame_info);
+
+    DEBUG("[esp_ieee802154] %s: frame=%p frame_info=%p\n",
+          __func__, frame, frame_info);
+
+    _rx_frame = frame;
+    _rx_frame_info = frame_info;
+
+    esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_INDICATION_RX_DONE);
+}
+
+void esp_ieee802154_transmit_sfd_done(uint8_t *frame)
+{
+    DEBUG("[esp_ieee802154] %s: frame=%p\n", __func__, frame);
+    if (IS_ACTIVE(_USE_TX_START)) {
+        esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_INDICATION_TX_START);
+    }
+}
+
+void esp_ieee802154_transmit_done(const uint8_t *frame, const uint8_t *ack,
+                                  esp_ieee802154_frame_info_t *ack_frame_info)
+{
+    DEBUG("[esp_ieee802154] %s: frame=%p ack_frame=%p ack_frame_info=%p\n",
+          __func__, frame, ack, ack_frame_info);
+
+    _ack_frame = ack;
+    _ack_frame_info = ack_frame_info;
+
+    esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
+}
+
+void esp_ieee802154_transmit_failed(const uint8_t *frame, esp_ieee802154_tx_error_t error)
+{
+    DEBUG("[esp_ieee802154] %s: frame=%p error=%u\n", __func__, frame, error);
+
+    _tx_error = error;
+
+    esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
+}
+
+void esp_ieee802154_energy_detect_done(int8_t power)
+{
+    DEBUG("[esp_ieee802154] %s: power %i\n", __func__, power);
+}
+
+void esp_ieee802154_setup(ieee802154_dev_t *dev)
+{
+    assert(dev);
+
+    DEBUG("[esp_ieee802154] %s: dev=%p\n", __func__, dev);
+
+    dev->driver = &esp_ieee802154_driver;
+    esp_ieee802154_dev = dev;
+}
+
+int esp_ieee802154_init(void)
+{
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+
+    esp_ieee802154_enable();
+    esp_ieee802154_reset_pending_table(true);
+    esp_ieee802154_reset_pending_table(false);
+    esp_ieee802154_set_rx_when_idle(true);
+    esp_ieee802154_receive();
+    esp_ieee802154_set_cca_mode(ESP_IEEE802154_CCA_MODE_ED);
+    esp_ieee802154_set_ack_timeout(3456);    /* should be a multiple of 16 */
```

Comment:
```suggestion
    esp_ieee802154_set_ack_timeout(216 * 16);    /* should be a multiple of 16 */
```

The compiler will optimize that away anyway, so why not write it in long form?

Reply:
My comment was not about the value itself, but rather about "it's more obvious that `216 * 16` is a multiple of 16 than the fact that `3456` is a multiple of 16".

Reply:
Okay.